proto/storage/image_v2.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+package storage;
+
+import "google/protobuf/timestamp.proto";
+import "storage/image.proto";
+
+option go_package = "./storage;storage";
+option java_package = "io.stackrox.proto.storage";
+
+// Next tag: 28
+message ImageV2 {
+  string id = 1; // @gotags: search:"Image ID,hidden" sql:"pk,type(uuid)"
+  string sha = 2; // @gotags: search:"Image Sha"
+  ImageName name = 3;
+
+  ImageMetadata metadata = 4;
+  ImageScan scan = 5; // @gotags: policy:"Image Scan"
+
+  ImageSignatureVerificationData signature_verification_data = 6;
+  ImageSignature signature = 7;
+
+  oneof set_components {
```

Comment:
I think I would add something to the names of these to make it obvious they are counts.

I think we need a comment in here saying why we are storing things like this that we can easily calculate.

Why are we using `oneof` here?  I admittedly don't fully understand `oneof` but this seems like an unnecessary extra layer.  On the surface just simply adding the field seems simpler.

Reply:
I just copied over from legacy image proto which used this pattern of using oneof for cve counts and such. I guess oneof allows those fields to be nil. But I don't see why that would have any benefit for these counts.